### DISCUSSION TITLE
Wrong volatile keyword usage in OHCI structures

### DISCUSSION THREAD

**Discussion by ua1arn** [hathach/tinyusb#2318] [General]
Proposed fix. I stand with unstable USB FLASH device enumeration
[patch.zip](https://github.com/hathach/tinyusb/files/13371898/patch.zip)
![image](https://github.com/hathach/tinyusb/assets/34999007/c156fff1-406f-4bae-bcef-ec27343e5bdd)

Also, on rv64 system (Allwinner F133-A) these fields have wrong size. uint32_t should be used instead of pointers.

**Comment by karlp** [comment]
Should this jsut be a PR or an issue directly? rather than a discussion thread?

**Comment by hathach** [comment]
late response, but yeah, it looks like a correct fix for current buffer pointer, though I think the buffer_end does not need volatile keyword since HC does not modify that. Since it is quite some time, I will make an PR myself, thank you for your proposed patch.

PS: PR is created #2479
